File: Easy/242-Valid Anagram/242.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if(s.size() == 0 && t.size() == 0)
            return true;
        else if(s.size() != t.size())
            return false;
        
        map<int, int> count;
        for(int i = 0; i < s.size(); ++i)
        {
            count[s[i]]++;
            count[t[i]]--;
        }
        for(map<int, int>::iterator it = count.begin(); it != count.end(); it++)
            if(it->second != 0)
                return false;
        return true;
    }
    /*
    void mergeSort(string &str, int front, int end)
    {
        if(front < end)
        {
            int mid = (front + end) / 2;
            mergeSort(str, front, mid);
            mergeSort(str, mid+1, end);
            merge(str, front, mid, end);
        }
    }
    
    void merge(string &str, int front, int mid, int end)
    {
        string left  = str.substr(front, mid-front+1);
        string right = str.substr(mid+1, end-mid);
        
        left  += 'z'+1;
        right += 'z'+1;
        
        int idx_left = 0, idx_right = 0;
        for(int i = front; i <= end; ++i)
        {
            if(left[idx_left] < right[idx_right])
                str[i] = left[idx_left++];
            else
                str[i] = right[idx_right++];
        }
    }
    */
    
};

File: Easy/242-Valid Anagram/242.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        if len(s) != len(t):
            return False
        elif len(s) == 0 and len(t) == 0:
            return True
        
        table = {}
        for i in range(0, len(s)):
            table[s[i]] = table.get(s[i], 0) + 1
            table[t[i]] = table.get(t[i], 0) - 1
        
        for letter in table:
            if table[letter] != 0:
                return False
        return True
        